sceptre/config/reader.py

```diff
@@ -429,49 +446,64 @@ def _render(self, directory_path, basename, stack_group_config):
-                    default=True,
-                ),
-                "loader": FileSystemLoader(abs_directory_path),
-                "undefined": StrictUndefined,
-            }
-            j2_environment_config = strategies.dict_merge(
-                default_j2_environment_config,
-                stack_group_config.get("j2_environment", {}),
+
+        if not path.isfile(path.join(abs_directory_path, basename)):
+            return
+
+        default_j2_environment_config = {
+            "autoescape": select_autoescape(
+                disabled_extensions=("yaml",),
+                default=True,
+            ),
+            "loader": FileSystemLoader(abs_directory_path),
+            "undefined": StrictUndefined,
+        }
+        j2_environment_config = strategies.dict_merge(
+            default_j2_environment_config,
+            stack_group_config.get("j2_environment", {}),
+        )
+        j2_environment = Environment(**j2_environment_config)
+
+        try:
+            template = j2_environment.get_template(basename)
+        except Exception as err:
+            raise SceptreException(
+                f"{Path(directory_path, basename).as_posix()} - {err}"
+            ) from err
+
+        self.templating_vars.update(stack_group_config)
+
+        try:
+            rendered_template = template.render(
+                self.templating_vars,
+                command_path=self.context.command_path.split(path.sep),
+                environment_variable=environ,
             )
-            j2_environment = Environment(**j2_environment_config)
-
-            try:
-                template = j2_environment.get_template(basename)
-            except Exception as err:
-                raise SceptreException(
-                    f"{Path(directory_path, basename).as_posix()} - {err}"
-                ) from err
-
-            self.templating_vars.update(stack_group_config)
-
-            try:
-                rendered_template = template.render(
-                    self.templating_vars,
-                    command_path=self.context.command_path.split(path.sep),
-                    environment_variable=environ,
+        except Exception as err:
+            message = f"{Path(directory_path, basename).as_posix()} - {err}"
+
+            if logging_level() == logging.DEBUG:
+                debug_file_path = self._write_debug_file(
+                    json.dumps(self.templating_vars), prefix="vars_"
                 )
-            except Exception as err:
-                raise SceptreException(
-                    f"{Path(directory_path, basename).as_posix()} - {err}"
-                ) from err
-
-            try:
-                config = yaml.safe_load(rendered_template)
-            except Exception as err:
-                raise ValueError(
-                    "Error parsing {}:\n{}".format(abs_directory_path, err)
+                message += f"\nTemplating vars saved to: {debug_file_path}"
```

Comment:
I spent some time testing the rest of this method, and found that exceptions raised here are just as confusing, therefore the code should also emit debug information in debug mode at this point.